### ISSUE TITLE
Schema composition does not work

### ISSUE THREAD

**Issue by gilles-yvetot** [Automattic/mongoose#9362]
**Do you want to request a *feature* or report a *bug*?**
*bug*
**What is the current behavior?**
Schema composition does not work
**If the current behavior is a bug, please provide the steps to reproduce.**
```javascript
const { Schema, model } = require('mongoose')

const StepSchema = new Schema(
  {
    view: {
      url: {
        type: String,
        trim: true,
      },
    },
  },
  { timestamps: true, discriminatorKey: `type` },
)

const ClickSchema = new Schema(
  [
    StepSchema,
    {
      view: {
        clickCount: {
          type: Number,
          default: 1,
          min: 1,
        },
      },
    },
  ],
  { discriminatorKey: `type` },
)

const Step = model(`Step`, StepSchema)

Step.discriminator(`click`, ClickSchema)

const doc = new Step({
  type: `click`,
  view: { url: `https://google.com` },
})

console.log('doc.view.url', doc.view.url)

```

**What is the expected behavior?**
`doc.view.url` to be defined
**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node 12.13.0 (*NOT* the browser environment)
Mongoose: 5.10.1
MongoDB: 4.2.8

**Comment by vkarpov15** [comment]
The issue here is that `view` from the 2nd schema definition overrides the `view` from the first schema definition. We'll see if we can add an option to merge these instead.

**Comment by gilles-yvetot** [comment]
Okay, got it. I was expected paths to be merged. If you cannot add the option, let me know and I'll hack around 👍

**Comment by vkarpov15** [comment]
Turns out there was a bug with our fix for #9042 that caused this issue. Fix will be in v5.10.3, thanks for reporting :+1:
